src/build/calibrate/calibrate_national.jl

```diff
@@ -0,0 +1,183 @@
+    + penalty_nokey * (
+        + sum(fd0[g,fd] for (g,fd) in set[:g,:fd] if d[:fd0][g,fd] == 0)
+        + sum(va0[va,s] for (va,s) in set[:va,:s] if d[:va0][va,s] == 0)
+        + sum(id0[g,s] for (g,s) in set[:g,:s] if d[:id0][g,s] == 0)
+        + sum(ys0[s,g] for (s,g) in set[:s,:g] if d[:ys0][s,g] == 0)
+        + sum(ms0[g,m] for (g,m) in set[:g,:m] if d[:ms0][g,m] == 0)
+        + sum(md0[m,g] for (m,g) in set[:m,:g] if d[:md0][m,g] == 0)
+        + sum(fs0[g] for g in G if d[:fs0][g] == 0)
+        + sum(a0[g]  for g in G if d[:a0][g]  == 0)
+        + sum(m0[g]  for g in G if d[:m0][g]  == 0)
+        + sum(x0[g]  for g in G if d[:x0][g]  == 0)
+        + sum(y0[g]  for g in G if d[:y0][g]  == 0)
+        )
+    )
+    
+    # Fix "certain parameters" to their original values: fs0, va0, m0.
+    fix!(calib, d, set, :va0, (:va,:s))
+    fix!(calib, d, set, [:fs0,:m0], :g)
+    
+    # Fix other/use sector output to zero.
+    fix!(calib, d, :ys0, [set[:oth,:use], G]; value=0)
+    
+    # --- OPTIMIZE AND SAVE RESULTS --------------------------------------------------------
+    JuMP.optimize!(calib)
+
+    return _calibration_output(calib, set, year; region=false)
+    # return calib
+end
+
+
+"""
+This function prepares
+- Filling zeros
+- 
+
+```math
+\\begin{aligned}
+x = max\\left\\{0, x\\right\\}
+fd_{g,fd} = max\\{0, fd_{g,fd}\\}
+```
+
+# Arguments
+- `d::Dict{Symbol,DataFrame}`: all input *parameters*
+- `set::Dict`
+- `year::Int` overwhich to calibrate
+
+# Returns
+- `d::Dict{Symbol, Dict}`: input *variables*. Input *variables* include all input parameters
+    with the exception of taxes (`ta0`,`tm0`,`ty0`).
+"""
+function _national_calibration_input(d, set;
+    lower_bound=SLiDE.DEFAULT_CALIBRATE_LOWER_BOUND,
+    upper_bound=SLiDE.DEFAULT_CALIBRATE_UPPER_BOUND,
+    zero_negative::Bool=true,
+)
+    variables = setdiff(SLiDE.list_parameters!(set,:calibrate), SLiDE.list_parameters!(set,:taxes))
+
+    # Fill zeros.
+    d = Dict(k => fill_zero(df; with=set) for (k,df) in d)
+    
+    # Handle negatives.
+    if haskey(d,:fd0)
+        SLiDE.zero_negative!(d[:fd0], :fd=>"pce")
+        SLiDE.zero_negative!(d[:fd0], :fd=>"C")
+    end
+    SLiDE.zero_negative!(d, setdiff(variables, [:fd0]))
+
+    # Set bounds.
+    SLiDE.set_lower_bound!(d, variables; factor=lower_bound, zero_negative=zero_negative)
+    SLiDE.set_upper_bound!(d, variables; factor=upper_bound, zero_negative=zero_negative)
+
+    d = Dict{Symbol,Dict}(k => convert_type(Dict, df) for (k,df) in d)
+    return d
+end
+
+
+function _national_calibration_input(d, set, year;
+    lower_bound=SLiDE.DEFAULT_CALIBRATE_LOWER_BOUND,
+    upper_bound=SLiDE.DEFAULT_CALIBRATE_UPPER_BOUND,
+    zero_negative::Bool=true,
+)
+    d = Dict(k => filter_with(df, (yr=year,); drop=true) for (k,df) in d)
+    d = _national_calibration_input(d, set;
+        lower_bound=lower_bound,
+        upper_bound=upper_bound,
+        zero_negative=zero_negative,
+    )
+    return d
+end
```

Comment:
Defined in calibrate_national.jl and not calibrate_io.jl because input is different for national calibration vs. energy calibration.